src/main/java/htsjdk/samtools/cram/structure/block/CompressibleBlock.java

```diff
@@ -0,0 +1,86 @@
+package htsjdk.samtools.cram.structure.block;
+
+/**
+ * Represents an arbitrary Block type.  It may be compressed and may be external.
+ * Contrast with {@link RawBlock} which can't be either of these
+ */
+public class CompressibleBlock extends Block {
+    /**
+     * Compression method that applied to this block's content.
+     */
+    private final BlockCompressionMethod method;
+
+    /**
+     * For external blocks, the external block Content ID
+     * Otherwise, NO_CONTENT_ID (0)
+     */
+    private final int contentId;
+
+    /**
+     * The content stored in this block, in uncompressed form
+     */
+    private final byte[] rawContent;
```

Comment:
alternate implementation: only store the compressed data, and uncompress on demand